### ISSUE TITLE
Parameterized tests fail with "Test framework quit unexpectedly" when trying to run a single test.

### ISSUE THREAD

**Issue by Colecf** [android/android-test#960]
### Description
When running a Parameterized instrumentation test with a custom name with AndroidJUnitRunner, specifying an individual test method causes the tests to fail with a "Test framework quit unexpectedly" error. This applies when using android studio's green arrows to launch a test, or when using `adb shell am instrument`.

![screenshot](https://raw.githubusercontent.com/Colecf/Android-Parameterized-Test-Issue-Repro/main/screenshot.png)
### Steps to Reproduce
Create a basic parameterized test with a custom parameter name and try running an individual method.

### Expected Results
One method should run and give a successful result.

### Actual Results
"Test framework quit unexpectedly" error.

### AndroidX Test and Android OS Versions
androidx.test.ext:junit:1.1.2. TOT build of android 12, and also the pixel 4 android 11 emulator downloaded from android studio.

### Link to a public git repo demonstrating the problem:
https://github.com/Colecf/Android-Parameterized-Test-Issue-Repro

**Comment by JuliaSullivanGoogle** [comment]
Confirmed fixed.

**Comment by Colecf** [comment]
Hi Julia, I just pushed a commit updating the reproduction git repository with all the latest versions of libraries, and it appears the issue hasn't been fully fixed. It no longer fails to run that test with an error, but the test results do not show up in the Android Studio UI, see the screenshot below.

![screenshot](https://github.com/Colecf/Android-Parameterized-Test-Issue-Repro/raw/main/screenshot_update_2022_7_26.png)

When testing this I also made sure to update to the latest public version of android studio, and updated my emulator to a pixel 5 api 33.
